### PR TITLE
Use Redux to manage loading of structs and their contents + SSR

### PR DESCRIPTION
**Pull request by subsocialdev**

### CODE REVIEW

File: src/rtk/features/posts/postsHooks.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const posts = useAppSelector(
+    state => selectPosts(state, { ids: postIds }),
+    
+    // TODO compare by (id + updated_at_block) instead of shallowEqual
+    shallowEqual
+  )
+
+  useSubsocialEffect(({ subsocial }) => {
+    dispatch(fetchPosts({ api: subsocial, ids: postIds }))
+  }, [ postIds, dispatch ])
```
Comment (olehmell) [inline src/rtk/features/posts/postsHooks.ts:19]: Maybe postIds.length?

File: src/rtk/features/posts/postsHooks.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { shallowEqual } from 'react-redux'
+import useSubsocialEffect from 'src/components/api/useSubsocialEffect'
+import { useAppDispatch, useAppSelector } from 'src/rtk/app/store'
+import { fetchPosts, FullPost, selectPosts } from 'src/rtk/features/posts/postsSlice'
+
+export const useFetchPosts = (postIds: EntityId[]): FullPost[] => {
+  const dispatch = useAppDispatch()
+
+  const posts = useAppSelector(
+    state => selectPosts(state, { ids: postIds }),
```
Comment (olehmell) [inline src/rtk/features/posts/postsHooks.ts:11]:
```suggestion
    state => loading ? [] : selectPosts(state, { ids: postIds }),
```

File: src/rtk/features/posts/postsSlice.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      owner = ownerByIdMap.get(post.owner)
+    }
+
+    result.push({ ...post, owner, space })
+  })
+  return result
+}
+
+function getUniqueSpaceIds (posts: NormalizedPost[]): EntityId[] {
+  const spaceIds = new Set<EntityId>()
```
Comment (olehmell) [inline src/rtk/features/posts/postsSlice.ts:77]: use in all place  `getUniqueIds` from `@subsocial/utils`?

**Comment by subsocialdev** [comment]
Already merged with another PR: https://github.com/dappforce/dappforce-subsocial-ui/pull/379

**Pull request closed without merging**